src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Tds/Tokens/DoneProc/DoneProcTokenParser.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClientX.IO;
+using Microsoft.Data.SqlClientX.Tds.Tokens.Done;
+
+namespace Microsoft.Data.SqlClientX.Tds.Tokens.DoneProc
+{
+    internal sealed class DoneProcTokenParser : TokenParser
+    {
+        public override async ValueTask<Token> ParseAsync(TokenType tokenType, TdsStream tdsStream, bool isAsync, CancellationToken ct)
+        {
+            ushort status = await tdsStream.TdsReader.ReadUInt16Async(isAsync, ct).ConfigureAwait(false);
+            DoneStatus doneStatus = (DoneStatus)status;
+
+            ushort currentCommand = await tdsStream.TdsReader.ReadUInt16Async(isAsync, ct).ConfigureAwait(false);
+
+            ulong rowCount;
+            if (tdsStream.TdsVersion > TdsVersion.V7_2)
```

Comment:
Why is TdsVersion stored with the stream? Should it not be closer to the connection/internal connection? 
Ideally streams should not hold the meaning that is associated with the connection like version information or other connection capabilities.

Reply:
Also in login we send a Tds version of 7.4 are we ever going to get into the `else` block?